__tests__/components/AmountInput.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { getComponent } from '@MOCKS/Components'
+import mosaicStore from '@/store/Mosaic'
+//@ts-ignore
+import AmountInput from '@/components/AmountInput/AmountInput.vue'
+// @ts-ignore
+import ErrorTooltip from '@/components/ErrorTooltip/ErrorTooltip.vue'
+import { ValidationProvider } from 'vee-validate'
+import { Tooltip } from 'view-design'
+import { mosaicsMock } from '@MOCKS/mosaics'
+const wrapper = getComponent(
+  AmountInput,
+  { mosaic: mosaicStore },
+  { mosaics: mosaicsMock },
+  { value: 0, mosaicHex: '519FC24B9223E0B4' },
+  { ErrorTooltip, ValidationProvider, Tooltip },
+)
+const vm = wrapper.vm as AmountInput
+describe('AmountInput', () => {
+  describe('the property amount of validationRules should be', () => {
+    test(`excluded:""|is_not:0|min_value:0|maxDecimals:6|max_value:9000000000000000`, () => {
+      wrapper.setProps({ mosaicHex: '519FC24B9223E0B4' })
+      expect(vm.validationRules.amount).toBe(
+        `excluded:""|is_not:0|min_value:0|maxDecimals:6|max_value:9000000000000000`,
+      )
+    })
+    test(`excluded:""|is_not:0|min_value:0|maxDecimals:5|max_value:9000000000000000`, () => {
+      wrapper.setProps({ mosaicHex: '534CD11F6D984B4B' })
+      expect(vm.validationRules.amount).toBe(
+        `excluded:""|is_not:0|min_value:0|maxDecimals:5|max_value:9000000000000000`,
+      )
+    })
+    test(`excluded:""|is_not:0|min_value:0|maxDecimals:3|max_value:9000000000000000`, () => {
+      wrapper.setProps({ mosaicHex: '119C8A107ACADD5F' })
+      expect(vm.validationRules.amount).toBe(
+        `excluded:""|is_not:0|min_value:0|maxDecimals:3|max_value:9000000000000000`,
+      )
+    })
+  })
```

Comment:
useless test, please **test the validator result** rather than its rules .... Rules can be changed depending on pages on which components are added, etc..